### PR TITLE
 Improve clone for COW SvPV when close to RefCnt Max

### PR DESCRIPTION
**Pull request by atoomic**
With the previous change one COWed SvPV was having
a CowRefcnt value set to the maximum value
( SV_COW_REFCNT_MAX or 255), the cloned SvPV was not COW
anymore.

This commit add an improved support for such SvPVs and preserve
the COW status on these strings, so (massive) cloning of the
same SvPV will improve memory usage and reduce duplicates.

Note: this is based on the previous PR #21

### CODE REVIEW

**Comment by atoomic** [comment]
Do not merge, this need more work

File: t/cow.t [hunk 1]
```diff
@@ -109,3 +108,41 @@ plan tests => 42;
...
+    my @a;
+    push @a, $str for 1 .. ( $max - 3 );
+    is cowrefcnt($str), $max - 2, "we are at max-2";
+
+    # now increase to max
+    push @a, clone($str); # fails on travis ???
+    is cowrefcnt($str), $max - 1, "we are now at max-1" or die;
+    is cowrefcnt( $a[-1] ), $max - 1, "we are now at max-1: a[-1]";
+
+    # no is time to bypass max
```
Comment (garu) [inline t/cow.t]: minor typo here, this should read "now" right?

File: t/cow.t [hunk 2]
```diff
@@ -109,3 +108,41 @@ plan tests => 42;
...
+    cmp_ok $max, '>', 2, "this should be greater than 2 :-)" or die;
+
+    # first let's do a stop just before max
+    my $str = "abcd";
+    my @a;
+    push @a, $str for 1 .. ( $max - 3 );
+    is cowrefcnt($str), $max - 2, "we are at max-2";
+
+    # now increase to max
+    push @a, clone($str); # fails on travis ???
```
Comment (garu) [inline t/cow.t]: what do you mean by "fails on travis"?

**Comment by garu** [comment]
I made some minor comments on the code, otherwise LGTM!

File: t/cow.t (hunk 2 above)
Comment (atoomic) [reply t/cow.t]: ok need to remove the comment :-) I mean that the regular perl version I'm using is behaving differently with COWPV with the notion of immortal cows... 
travis was just the stack of regular Perl

**Comment by atoomic** [comment]
ready for round 2, the two changes were only comment fixup

File: t/cow.t (hunk 1 above)
Comment (atoomic) [reply t/cow.t]: s{no}{now}

**Pull request merged**